src/App.tsx

```diff
@@ -15,36 +15,106 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortType {
+  alphabetically = 'alphabetically',
+  length = 'length',
+}
+
+function getPreparedGoods(
+  goods: string[],
+  { sortField, isReversed }: { sortField: SortType; isReversed: boolean },
+) {
+  const preparedGoods = [...goods];
+
+  if (sortField) {
+    switch (sortField) {
+      case 'alphabetically':
+        preparedGoods.sort((good1, good2) => good1.localeCompare(good2));
+        break;
+      case 'length':
+        preparedGoods.sort((good1, good2) => good1.length - good2.length);
+        break;
+      default:
+        break;
+    }
+  }
+
+  if (isReversed) {
+    preparedGoods.reverse();
+  }
+
+  return preparedGoods;
+}
+
 export const App: React.FC = () => {
+  const [sortField, setSortField] = useState('');
+  const [isReversed, setIsReversed] = useState(false);
+
+  const visibleGoods = getPreparedGoods(goodsFromServer, {
+    sortField: sortField as SortType,
```

Comment:
Casting `sortField` to `SortType` without validation might lead to runtime errors if `sortField` is not a valid `SortType`. Ensure that `sortField` is always a valid `SortType` or handle invalid cases in `getPreparedGoods`.